app/modules/map/map.redux.js

```diff
@@ -0,0 +1,38 @@
+import { Record, Map } from 'immutable';
+import { createReducer, createActions } from 'reduxsauce';
+import { PIN_MODE } from './map.constants';
+
+export const { Types: MapTypes, Creators: MapActions } = createActions({
+  changeMode: ['mode'],
+  changePosition: ['lat', 'long'],
+  changeLocationPermissions: ['permitted'],
+}, { prefix: 'MAP_' });
+
+const MapRecord = new Record({
+  mode: PIN_MODE,
+  position: Map({
+    lat: 52.414263,
+    long: 16.920183,
+  }),
+  hasLocationPermissions: false,
+});
+
+export const INITIAL_STATE = new MapRecord({});
+
+export const changeMode = (state = INITIAL_STATE, { mode }) => state
+  .set('mode', mode);
+
+export const changePosition = (state = INITIAL_STATE, { lat, long }) => state
+  .setIn(['position', 'lat'], lat)
```

Comment:
chyba łatwiej używać merge'a przy ustawianiu wielu propów, ale to tylko taka kosmetyczna uwaga